### ISSUE TITLE
"cannot find job file" when inspecting datum after migration from 1.9.0 to 1.10.0

### ISSUE THREAD

**Issue by pappasilenus** [pachyderm/pachyderm#4671]
**What happened?**:

I created a pachyderm 1.9.0 cluster consisting of the following examples
*  opencv
* object-detection
* hyperparameter
* gatk

I then backed up the cluster using pachctl extract.

I deleted all the objects using pachctl delete all, and deleted the cluster using pachctl undeploy --all.

I deployed 1.10.0 and then restored the extracted backup to migrate the cluster.

There was a failed job at the top of the list for the `detect` pipeline (along with many others).  I was able to inspect the job fine, but inspecting the datum resulted in the error "cannot find job file".  Stats were turned on for this pipeline.

**What you expected to happen?**:

I expected to be able to inspect the datum.

**How to reproduce it (as minimally and precisely as possible)?**:

(someday I'll post Makefile changes that make this easier.)

1. deploy 1.9.0 on a clean cluster
2. check out source and cd to examples, deploy opencv using "make opencv"
3. add images3.txt and images4.txt to the images repo.
4. In the makefile for examples, comment out the creation of the images repo for the object-detection target.  It'll share the repo for opencv.
5. edit  ml/object-detection/detect.json, enabling stats
6. deploy object detection with "make object-detection".
7. edit ml/hyperparameter/model.json, changing the name to "hyperparameter-model"
8. edit  ml/hyperparameter/select.json, changing the repo of the "model" input to "hyperparameter-model" and adding a name attribute `"name": model`
9. edit  ml/hyperparameter/test.json, changing the repo of the "model" input to "hyperparameter-model" and adding a name attribute `"name": model`
10. deploy hyperparameter with "make hyperparameter"
11. deploy gatk with "make gatk".
12. wait for all the jobs to run to completion.
13. extract the back using pachctl extract to a file
14. pachctl delete all
15. pachctl undeploy --all
16. deploy 1.10.0
17. pachctl restore from the file from 13
18. See failed jobs, Try inspect datum on datum that's fingered in a failed job in detect

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}
```
- Pachyderm CLI and pachd server version (use `pachctl version`):
```
COMPONENT           VERSION             
pachctl             1.10.0              
pachd               1.10.0-rc1   
```
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s):
docker-desktop on macOS
6G ram assigned
- OS (e.g. from /etc/os-release): macOS Catalina 10.15.3 (19D76)
- Others:

**Comment by pappasilenus** [comment]
I've created a PR that adds some Makefile targets to make reproducing this easier:

https://github.com/pachyderm/pachyderm/pull/4678

**Comment by nitinjainsj** [comment]
@pappasilenus I think this one is fixed PR #4680
